src/libertem/io/dataset/hdf5.py

```diff
@@ -300,25 +485,34 @@ def _get_tiles_with_roi(self, roi, dest_dtype):
         frames_read = 0
         start_at_frame = self.slice.origin[0]
         frame_offset = np.count_nonzero(flat_roi[:start_at_frame])
-
+        skip_tile = False
         indices = _roi_to_nd_indices(roi, self.slice_nd)
 
         with self.reader.get_h5ds() as h5ds:
             for idx in indices:
-                tile_slice = Slice(
-                    origin=(frames_read + frame_offset,) + sig_origin,
-                    shape=result_shape,
-                )
-                tile_data = h5ds[idx].reshape(result_shape)
-                self._preprocess(tile_data, tile_slice)
-                yield DataTile(
-                    tile_data,
-                    tile_slice=tile_slice,
-                    # there is only a single slice in the tiling scheme, so our
-                    # scheme_idx is constant 0
-                    scheme_idx=0,
-                )
-                frames_read += 1
+                if self.meta.sync_offset != 0:
+                    idx_pos_with_offset = (
+                        np.ravel_multi_index(idx, self.meta.shape.nav) + self.meta.sync_offset
+                    )
+                    if 0 <= idx_pos_with_offset < int(np.prod(self.meta.shape.nav)):
+                        idx = np.unravel_index(idx_pos_with_offset, self.meta.shape.nav)
+                    else:
+                        skip_tile = True
```

Comment:
Can't this be a `break` instead? Once `skip_tile` is set, it's never unset and no further tiles are yielded